src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -1723,8 +1725,15 @@ UniqueJobId(void)
 	jobIdDatum = DirectFunctionCall1(nextval_oid, sequenceIdDatum);
 	jobId = DatumGetInt64(jobIdDatum);
 
+	/*
+	 * Add the local group id information to the jobId to
+	 * prevent concurrent jobs on different groups to conflict.
+	 */
+	localizedJobId = jobId | (localGroupId << 32);
```

Comment:
Oh. 2^32 is the maximum value, so it's actually off by one. Anyway, I'm inclined to just fix that by moving away from using a sequence.